cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/internal/SSAConstruction.qll

```diff
@@ -339,14 +339,14 @@ private module Cached {
   }
 
   cached
-  Opcode getInstructionOpcode(Instruction instr) {
-    result = getOldInstruction(instr).getOpcode()
+  predicate getInstructionOpcode(Opcode opcode, Instruction instr) {
```

Comment:
Please add a comment to this predicate, explaining why the parameter order matters. Otherwise there's a risk that someone changes the order back in the future.

The predicate name is also slightly off now. Would `instructionHasOpcode` be better? I'm not really sure.

Reply:
I agree the name is slightly off now. I don't think `instructionHasOpcode` is better, though. Grep'ping `predicate [a-z]+Has[A-Z][a-z]+` on our code makes me think that `xHasY` is also given the parameter ordering `(x, y)`. So it should really be something like `opcodeGivesInstruction` (even though I also don't like that).